### ISSUE TITLE
Windows: "Cannot read property length of null" when parsing SQLite path connection string

### ISSUE THREAD

**Issue by Seikho** [knex/knex#3743]
```sh
seikho@PC MINGW64 /c/projects/project (dev)
$ node .
C:\projects\project\node_modules\knex\lib\util\parse-connection.js:9
  const isDriveLetter = isWindows && protocol.length === 2;
                                              ^

TypeError: Cannot read property 'length' of null
    at parseConnectionString (C:\projects\project\node_modules\knex\lib\util\parse-connection.js:9:47)
```

The changes introduced in https://github.com/knex/knex/commit/c14252d907855cee310db7b57fe37a776fd4424a throws a type error when parsing a SQLite connection string with a file path. Reverting to 0.22.8 fixes the issue. Not checking `protocol` exists before using it causes the error.

# Environment

Knex version: 0.22.9 - current
Database + version: SQLite
OS: Windows 10 stable

Select applicable template from below.
If issue is about oracledb support, tag @ atiertant. For MSSql tag @ smorey2.
Rest of dialects don't need tags.
If issue is about TypeScript definitions, tag @ lorefnon.

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

2. Error message

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

# Feature discussion / request

1. Explain what is your use case

2. Explain what kind of feature would support this

3. Give some API proposal, how the feature should work

# Missing / erroneus documentation

Send issue to documentation repo, or fix it and send PR https://github.com/knex/documentation

# Questions about how to use knex

GitHub issues are for knex development. Please send questions how to use knex to
Stack Overflow or ask about it in gitter chat.

**Comment by kibertoad** [comment]
Thanks for reporting this, I'll make a fix shortly.

**Comment by kibertoad** [comment]
Released in 0.20.13
